GeoLib/Polygon.cpp

```diff
@@ -125,9 +125,9 @@ std::vector<GeoLib::Point> Polygon::getAllIntersectionPoints(
 {
     std::vector<GeoLib::Point> intersections;
     GeoLib::Point s;
-    for (auto seg_it(begin()); seg_it != end(); ++seg_it)
+    for (auto && seg_it : *this)
```

Comment:
`this` is a Polygon, which is a Polyline, which has begin() and end() returning a Polyline::SegmentIterator, a value, or more correctly a prvalue. This prvalue is captured in the rvalue reference 'seg_it'. This is what clang-tidy saw.

Timings:
original 0.02190
for-range-loop with auto 0.02128
for-range-loop with auto const& or auto&& 0.02111